action.py

```diff
@@ -53,6 +53,34 @@ def _log(msg):
     print(msg, file=sys.stderr)
 
 
+def _download_ref_asset(ext):
+    repo = os.getenv('GITHUB_REPOSITORY')
+    ref = os.getenv("GITHUB_REF")
+
+    artifact = f"{os.getenv('GITHUB_REF_NAME')}-signed{ext}"
+
+    # GitHub supports /:org/:repo/archive/:ref<.tar.gz|.zip>.
+    # XX: will this work in Windows runners?
+    curl_status = subprocess.run(
+        ["curl",
+         "-f",
+         "-L",
+         "-o", str(artifact),
+         f"https://github.com/{repo}/archive/{ref}{ext}"],
+        text=True,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.STDOUT,
+        # do not pass environment to curl
+    )
+    _debug(curl_status.stdout)
```

Comment:
LGTM for the first blush, but in the interest of "native" requests let's use `requests`. 

Something roughly like:

```python
resp = requests.get("https://...")
resp.raise_for_status()

with Path(artifact).open("wb") as io:
    io.write(resp.content)
```

(You'll need to add `requests` to the `requirements.txt` as well, most likely.)